src/render.js

```diff
@@ -411,9 +412,57 @@ class App extends Preact.Component {
 
   openAboutPage = () => ipcRenderer.send(OPEN_ABOUT_PAGE);
 
+  /**
+   * @param {string} input
+   */
+  filterBy = (input) => {
+    // Helpers
+    const unfiltered = () => true;
+    /** @type {string} */
+    const isOfType = (type) => (entry) => entry.type === type;
+    /** @type {string} */
+    const isOfMimeType = (type) => (entry) => entry._type === type;
+
+    // No input - return everything
+    if (!input) return unfiltered;
+
+    // No label - search by text (default)
+    if (!input.includes(":"))
+      return (entry) => isOfType("text")(entry) && entry.value.includes(input);
+    else {
+      const [label, value] = input
+        ?.split(":") // split on ":"
+        ?.map((element) => element.trim()); // trim whitespaces
+
+      if (label === "text")
+        //TODO improve searching
+        return (entry) =>
+          isOfType("text")(entry) && entry.value.includes(value);
+
+      if (label === "image") {
+        //TODO more MIME types
+        if (value === "image/png")
+          return (entry) =>
+            isOfType("image")(entry) && isOfMimeType("image/png")(entry);
```

Comment:
Maybe forward the MIME type after the `/` immediately? This would support anything the user types in. 

```js
if (label === "image") {
  return (entry) => isOfType("image")(entry) && isOfMimeType(`image/${value}`)(entry);
}
```

This would allow the user to type `image: png` and it'd just work! 😄